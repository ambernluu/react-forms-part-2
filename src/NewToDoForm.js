import React, {useState} from 'react';
import { v4 as uuid } from 'uuid';

const NewToDoForm = ({addToDo}) => {
    const [task, setTask] = useState("");

    const handleChange = e => {
        setTask(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addToDo({ task, id: uuid() });
        setTask("");
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>To Do List</h1>
            <label htmlFor="task">Task </label>
            <input
                id="task"
                type="text"
                name="task"
                placeholder="Add a new task"
                onChange={handleChange}
                value={task}
            />

            
            <button>Update To Do List!</button>
        </form>
    )
}

export default NewToDoForm;