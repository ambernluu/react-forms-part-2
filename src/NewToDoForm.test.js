import { render, screen } from '@testing-library/react';
import NewToDoForm from './NewToDoForm';

it("renders without crashing", () => {
    render(<NewToDoForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewToDoForm />);
    expect(asFragment()).toMatchSnapshot();
});