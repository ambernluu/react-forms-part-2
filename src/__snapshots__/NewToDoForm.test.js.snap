// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <h1>
      To Do List
    </h1>
    <label
      for="task"
    >
      Task 
    </label>
    <input
      id="task"
      name="task"
      placeholder="Add a new task"
      type="text"
      value=""
    />
    <button>
      Update To Do List!
    </button>
  </form>
</DocumentFragment>
`;
